frontend/src/components/UserDashboard/UserFlights/SearchFlight.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect } from "react";
+import Select from "react-select";
+import "./searchflight.css";
+
+const SearchFlight = ({
+  filters,
+  locations,
+  onInputChange,
+  onSearch,
+  onFilterCategory,
+}) => {
+  const [selectedSource, setSelectedSource] = useState(null);
+  const [selectedDestination, setSelectedDestination] = useState(null);
```

Comment:
- Instead of using separate useState hooks for source, destination, sourceDate, and destinationDate, you can combine them into a single useState object to simplify state management and reduce redundancy.


- Using this approach, you would maintain a single state object, e.g.,
`const [filters, setFilters] = useState({ source: null,  destination: null, sourceDate: "",  destinationDate: "" });,`
and create a dynamic handleChange function, such as
`const handleChange = (key, value) => { setFilters((prev) => ({ ...prev, [key]: value })); onInputChange(key, value); };.`
This method allows you to update any field by passing its key and value, improving scalability and code clarity.

- This approach aligns with React's official documentation under "Updating Objects and Arrays in State" ([reference link](https://react.dev/reference/react/useState#examples-objects)) and is demonstrated in the provided image.